app/controllers/proposals_controller.rb

```diff
@@ -98,6 +99,13 @@ def proposal_params
                                      speakers_attributes: [:bio, :user_id, :id])
   end
 
+  def require_invite
+    unless current_user.invitations.where(state: 'pending').includes(proposal_id: @proposal.id)
+      redirect_to root_path
+      flash[:danger] = 'You are not an invited speaker for the proposal you are trying to access.'
+    end
+  end
+
```

Comment:
I wonder if this is always returning true then. . . I tried it in the console and this is what I see:

```
pry(main)> u.invitations.where(state: 'pending').includes(proposal_id: p.id)
  Invitation Load (0.9ms)  SELECT "invitations".* FROM "invitations" WHERE "invitations"."user_id" = $1 AND "invitations"."state" = $2  [["user_id", 1], ["state", "pending"]]
=> []
```

So that query is always returning at least `[]`, which is actually considered truthy:

```
pry(main)> if []
pry(main)*   "yes"
pry(main)* end  
=> "yes"
```

In order to see if anything matches you can add `.any?` or `.first` to the query. I also think the use of `includes` might be wrong. . . I added a comment about that too.